### PR TITLE
`azurerm_sentinel_alert_rule_fusion` - support for `source`block

### PR DESCRIPTION
**Pull request by magodo**
Add new property `source_setting` for `azurerm_sentinel_alert_rule_fusion`.

In my first try, I've omit the `enabled` properties in the schema (see: https://github.com/hashicorp/terraform-provider-azurerm/commit/383de98afb4786c063a6048982478bfdb37f7248). Due to the API issue https://github.com/Azure/azure-rest-api-specs/issues/21387, it won't work.

## Test

```shell
💢  TF_ACC=1 go test -v -timeout=20h ./internal/services/sentinel -run=TestAccSentinelAlertRuleFusion_
=== RUN   TestAccSentinelAlertRuleFusion_basic
=== PAUSE TestAccSentinelAlertRuleFusion_basic
=== RUN   TestAccSentinelAlertRuleFusion_disable
=== PAUSE TestAccSentinelAlertRuleFusion_disable
=== RUN   TestAccSentinelAlertRuleFusion_sourceSetting
=== PAUSE TestAccSentinelAlertRuleFusion_sourceSetting
=== RUN   TestAccSentinelAlertRuleFusion_requiresImport
=== PAUSE TestAccSentinelAlertRuleFusion_requiresImport
=== CONT  TestAccSentinelAlertRuleFusion_basic
=== CONT  TestAccSentinelAlertRuleFusion_sourceSetting
=== CONT  TestAccSentinelAlertRuleFusion_requiresImport
=== CONT  TestAccSentinelAlertRuleFusion_disable
--- PASS: TestAccSentinelAlertRuleFusion_basic (185.42s)
--- PASS: TestAccSentinelAlertRuleFusion_requiresImport (230.13s)
--- PASS: TestAccSentinelAlertRuleFusion_sourceSetting (234.34s)
--- PASS: TestAccSentinelAlertRuleFusion_disable (258.68s)
PASS
```

### CODE REVIEW

File: internal/services/sentinel/sentinel_alert_rule_fusion_resource.go [hunk 1]
```diff
@@ -62,6 +62,62 @@ func resourceSentinelAlertRuleFusion() *pluginsdk.Resource {
 				Optional: true,
 				Default:  true,
 			},
+
+			"source_setting": {
```
Comment (katbyte) [inline internal/services/sentinel/sentinel_alert_rule_fusion_resource.go:66]: setting is redundant?
```suggestion
			"source": {
```

File: internal/services/sentinel/sentinel_alert_rule_fusion_resource.go [hunk 2]
```diff
@@ -62,6 +62,62 @@ func resourceSentinelAlertRuleFusion() *pluginsdk.Resource {
...
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"enabled": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+							Default:  true,
+						},
+						"source_sub_type": {
```
Comment (katbyte) [inline internal/services/sentinel/sentinel_alert_rule_fusion_resource.go:83]: source is part of the block name?
```suggestion
						"sub_type": {
```

File: internal/services/sentinel/sentinel_alert_rule_fusion_resource.go [hunk 3]
```diff
@@ -62,6 +62,62 @@ func resourceSentinelAlertRuleFusion() *pluginsdk.Resource {
...
+										Type:         pluginsdk.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringIsNotEmpty,
+									},
+									"enabled": {
+										Type:     pluginsdk.TypeBool,
+										Optional: true,
+										Default:  true,
+									},
+									"enabled_severities": {
```
Comment (katbyte) [inline internal/services/sentinel/sentinel_alert_rule_fusion_resource.go:98]: should this be
```suggestion
									"severities_enabled": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/sentinel/sentinel_alert_rule_fusion_resource.go (hunk 3 above)
Comment (magodo) [reply internal/services/sentinel/sentinel_alert_rule_fusion_resource.go:98]: @katbyte We used `foo_enabled` as a boolean property, whilst this one is an array of enabled severities.

**Comment by magodo** [comment]
@katbyte Properties are renamed, please take another look!

File: internal/services/sentinel/sentinel_alert_rule_fusion_resource.go (hunk 3 above)
Comment (katbyte) [reply internal/services/sentinel/sentinel_alert_rule_fusion_resource.go:98]: yes but putting it in front doesn't make sense. looking at the api closer, should this just be
```suggestion
									"severities": {
```
as it is the severities for this sub_type, so `sub_type.enable = true` means enable then, `sub_type.enabled = false` would actually disable them?

so maybe
```suggestion
									"severities_allowed": {
```
or
```suggestion
									"severities_filter": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/sentinel/sentinel_alert_rule_fusion_resource.go (hunk 3 above)
Comment (magodo) [reply internal/services/sentinel/sentinel_alert_rule_fusion_resource.go]: @katbyte I've renamed it to `severities_allowed` now.

**Review by katbyte — APPROVED** [review]
THanks @magodo  LGTM :atom:

**Pull request merged**